### ISSUE TITLE
Add a Flatten layer on ResNet50(notop) and get an error

### ISSUE THREAD

**Issue by JWarlock** [fchollet/keras#6198]
I am trying to add a Flatten layer, a Dense layer(relu) and a Dense layer(softmax) on ResNet50 for a multi-classification task using Keras 2.0.2 Theano 0.9.0 py2.7 on Win10.Here is my code:
    def create_model():
    base_model = ResNet50(include_top=False, weights=None,
                      input_tensor=None, input_shape=(3,224,224),
                      pooling=None)
    base_model.load_weights(weight_path+'/resnet50_weights_th_dim_ordering_th_kernels_notop.h5')
    x=base_model.output
    x = Flatten()(x)
    x=Dense(128,activation='relu',kernel_initializer='random_normal',
        kernel_regularizer=regularizers.l2(0.1),
        activity_regularizer=regularizers.l2(0.1))(x)
    x=Dropout(0.3)(x)
    y = Dense(8, activation='softmax')(x)
    model = Model(base_model.input, y)
    for layer in base_model.layers:
        layer.trainable = False
      model.compile(optimizer='adadelta',
                  loss='categorical_crossentropy')
    return model

But I always get following error message:
The shape of the input to "Flatten" is not fully defined (got (2048, None, None). Make sure to pass a complete "input_shape" or "batch_input_shape" argument to the first layer in your model.

I have set image_dim_ordering:
from keras import backend as K
K.set_image_dim_ordering('th')

And here is my Keras.json file:
{
    "backend": "theano",
    "image_data_format": "channels_first",
    "epsilon": 1e-07,
    "floatx": "float32"
}

**Comment by fchollet** [comment]
Here is a reproduction script (this is what you should have posted):

```python
from keras.applications.resnet50 import ResNet50
from keras import backend as K
from keras.models import Model
from keras.layers import Flatten, Dense

K.set_image_data_format('channels_first')

base_model = ResNet50(include_top=False, weights=None,
                      input_shape=(3, 224, 224))
x = base_model.output
x = Flatten()(x)
y = Dense(8, activation='softmax')(x)
model = Model(base_model.input, y)
```

This completes as expected (with both backends).

**Comment by JWarlock** [comment]
However, if I add this line:
`base_model.load_weights(weight_path+'/resnet50_weights_th_dim_ordering_th_kernels_notop.h5')`
before 
`x = Flatten()(x)`
I get this error:
The shape of the input to "Flatten" is not fully defined (got (2048, None, None). Make sure to pass a complete "input_shape" or "batch_input_shape" argument to the first layer in your model.

if I do not load_weights,everthing works fine.
